### PR TITLE
Configurable refresh time, circuits Timer

### LINKED ISSUE #2: Clean up spinlock in refresh from Avalon
Right now the avalon_refresh_function is called repeatedly by the resilient_run command, and simply returns nothing unless it's been 60 minutes. Unfortunately, that's effectively spinlocking one CPU on the system running resilient_run. IBM has asked us to not do that, and use sleeps instead.

### LINKED ISSUE #3: Make refresh time configurable
IBM has asked us to make the refresh time called in the avalon_refresh_function configurable by the user in the config file, rather than hardcoded to 60 minutes.

### PR DESCRIPTION
**Pull request by vkantchev**
* Configurable refresh time - closes #3
* Implement auto refresh with circuits Timer - closes #2

### CODE REVIEW

File: avalon/components/avalon_component.py [hunk 1]
```diff
@@ -91,10 +106,68 @@ def handle_avalon_push_artifact(self, event, *args, **kwargs):
...
+
+
+    def _start_auto_refresh(self, incident):
+        # Get the interval from the avalon_auto_refresh_time field 
+        refresh_interval_minutes = self.actions.res.incident_get_avalon_auto_refresh_time(incident)
+
+        # create timer
+        incident_id = incident["id"]
+        auto_refresh_event = auto_refresh(incident_id)
+        timer = Timer(60 * refresh_interval_minutes, auto_refresh_event, self.channel, persist=True)
```
Comment (g-clef) [inline avalon/components/avalon_component.py:168]: May need to check that `refresh_interval_minutes` is not None. I'm getting that error on my test instance.

Comment (vkantchev) [reply avalon/components/avalon_component.py:168]: Now defaults to 60 minutes if not set in IBM Resilient

**Comment by g-clef** [comment]
Within Resilient-run, I'm getting another error now:
```
2019-04-16 10:36:01,108 INFO [actions_component] Event: <avalon_refresh[] (id=2, workflow=avalon_refresh, user=aaron@kingandunion.com) 2019-02-01 20:38:31.118000> Channel: functions.avalon_refresh
2019-04-16 10:36:01,110 ERROR [action_message] FunctionResult must be a dictionary. 'NoneType' may cause the workflow to fail. ```

**Comment by vkantchev** [comment]
Did you run:

```bash
resilient-circuits customize
```

**Comment by vkantchev** [comment]
Also delete old workflows. I think `resilient-circuits customize` does not delete stuff.  The old refresh workflow is no longer needed.

There should not be any workflows defined:

image

**Pull request merged**